src/usart/tx.rs

```diff
@@ -278,6 +278,38 @@ where
     }
 }
 
+impl<I, W, Function> Tx<I, Enabled<W>, CtsThrottle<Function>>
+where
+    I: Instance,
+    W: Word,
+{
+    /// Disable throttling via CTS signal
+    ///
+    /// Configure the transmitter to ignore the CTS signal. Returns the SWM
+    /// function for the CTS signal, so it can be reused to enable CTS
+    /// throttling again, or for something else.
+    pub fn disable_cts_throttling(
+        self,
+    ) -> (Tx<I, Enabled<W>, NoThrottle>, Function) {
+        interrupt::free(|_| {
```

Comment:
Not sure if you're planning to change that later, but this currently isn't needed, since we're only changing this in the Usart enable function and in `Tx` stuff. I also prefer if users have the option to supply the `cs` separately, in case they know better (e.g. when using rtic) & not disabling interrupts, but there are definitely pros and cons to that approach as well.

Reply:
Even if we could do without a critical section here, I don't think we should, unless we're very sure the API is no longer going to change much. It's way too easy to make a change elsewhere and invalidate the assumptions made here.

Having the user supply the critical section token is an interesting idea. I'll look into it later. I'm a bit concerned how the move required by this API is going to interact with an `interrupt::free` closure at the call site, but I think it shouldn't be a problem.